src/lib/components/ScrollSentinel.svelte

```diff
@@ -1,21 +1,41 @@
 <script lang="ts">
   import { layoutContentTopHidden } from "$lib/stores/layout.store";
-  import { onMount } from "svelte";
+  import { onMount, tick } from "svelte";
+  import { isNullish } from "@dfinity/utils";
 
   // The ScrollSentinel component should be placed right before the content
   // inside the scrollable container.
-  export let scrollContainer: HTMLElement;
+
+  interface Props {
+    scrollContainer: HTMLElement;
+  }
+
+  let { scrollContainer }: Props = $props();
 
   // To observe when the top leaves the view
-  let element: HTMLElement;
+  let element = $state<HTMLElement | undefined>();
 
   onMount(() => {
-    const observer = new IntersectionObserver(
-      ([entry]) => layoutContentTopHidden.set(!entry.isIntersecting),
-      { root: scrollContainer, threshold: 0 },
-    );
-    observer.observe(element);
-    return () => observer.disconnect();
+    let observer: IntersectionObserver | undefined = undefined;
+
+    const setObserver = async () => {
+      await tick(); // wait for DOM + bind:this to complete
```

Comment:
Mmm that's recommended by the doc? In `InfiniteScroll` with use an `$effect`, would that work here as well?

Reply:
Kind of, like in InfiniteScoll https://github.com/dfinity/gix-components/blob/3f28aa0b44ee668ea143e1928d518def7401eebb/src/lib/components/InfiniteScroll.svelte#L42C3-L45C5

```
const observer: IntersectionObserver = new IntersectionObserver(
    onIntersection,
    options,
  );
  
  $effect(() =>. {
     observer.unsubribe
      
      observer.subscribe
  
  )
  
  ondestory(unsubscribe)
  ```

Reply:
Looks great, thanks!